cax/tasks/filesystem.py

```diff
@@ -163,6 +163,108 @@ def each_run(self):
 
             break
 
+
+class AddSize(Task):
+
+    """ Evaluate the file size of a raw data and check if
+        the number of files in the raw data directory match
+        with the number of events recorded
+    """
+    def __init__(self):
+        Task.__init__(self)
+
+    def each_run(self):
+
+        # Check of data info
+        if 'data' not in self.run_doc:
+           self.log.debug("Data not found Name: %s " %(self.run_doc['name']) )
+           return
+
+        # Check of Trigger info
+        if 'trigger' not in self.run_doc:
+           self.log.debug("Trigger not found Name: %s " %(self.run_doc['name']) )
+           return
+
+        run_number = self.run_doc['number']
+        run_name = self.run_doc['name']
+
+
+        trigger = self.run_doc['trigger']
+        nevents = trigger['events_built']
+
+        for data_doc in self.run_doc['data']:
+
+            _host = data_doc['host']
+            _location = data_doc['location']
+            _type = data_doc['type']
+            _status = data_doc['status']
+
+            try:
+                if _type ==  "raw" and _status == "transferred":
+                    self.log.debug("host: %s  location: %s  type: %s" %(_host,_location,_type ) )
+
+
+                    # Check if the number of files match with the number of events
+                    completeness = False
+                    raw_size = 0
+                    nfiles = len(os.listdir(_location))-4
+                    ents = int(nevents)/1000.
```

Comment:
@lucrlom As you mentioned, the number of events per file is not constant throughout our data taking.  Do you really need this check?  Shouldn't the status check of ```transferred``` above be sufficient?  If not, we really shouldn't be tagging those files as ```transferred```, and this points to another problem.  Have you actually observed this error case yet?

Reply:
I did this because in the past I saw some raw data that was not complete on midway. Maybe because oldest code of Rucio didn't perform all present check. Maybe now is more reliable than before.

Reply:
ok, this kind of error should be reported to @XeBoris, documented, and fixed upstream.

if you believe you still need to keep this check here, then i think you can use the DB key:
```
"reader": {
    "ini": {
        "trigger_config_override": {
            "Zip": {
                "events_per_file": 100
            }
        }
    }
}
```
(I just found searching through a RunsDB JSON.)

Reply:
@lucrlom Can you be more specific about past raw data on Midway? Do you mean that we missed individual zip files? Ruciax compares the file name of the uploaded zip files after upload again to the registered zip files of a raw data set and therefore a raw data set is only marked as "transferred" if this test is positive.